plugins/gatsby-theme-iterative-docs/src/components/Documentation/Markdown/index.tsx

```diff
@@ -58,11 +60,17 @@ const Details: React.FC<{ slugger: GithubSlugger }> = ({
         ? cur?.props?.children?.toString()
         : '')
   }, '')
-
-  let slug = slugger.slug(title)
-  slug = slug.replaceAll('️', '').replaceAll('ℹ', '')
-  const id = slug.replace(/(^\-+|\-+$)/g, '')
-
+  try {
+    id = id
+      ? useMemo(() => slugger.slug(id), [id])
+      : useMemo(() => slugger.slug(title), [title])
```

Comment:
Does `useMemo` do anything here? This also seems to sort of break a rule of hooks that all hooks must always be called, but only technically because I having the same hook always called means the whole "hooks are preserved in the order they're called" thing that necessitates the rule is satisfied.

Reply:
There was an issue with the slugger function throwing error for the same title during rerenders. So, this would skip generating slug for the same title or id during rerenders. Since this is component level this will not have side effects with other `<details>` component titles. 

> all hooks must always be called

So, if you mean if one gets another will not get called. In that case, we can have the conditional statement inside the useMemo function.

Reply:
> So, if you mean if one gets another will not get called. In that case, we can have the conditional statement inside the useMemo function.

Yep, that's usually how `useMemo` calls with conditionals are handled.

Reply:
> Your explanation makes me think of how it's an interesting case that the page could error out if the writer goes back and forth on an id, or moves the same one around. It's enough of an edge case to be overlooked, especially since it can be cleared with a page refresh, but it's something worth looking into for the future

Umm, it errors out if we don't use the `useMemo` here. And, it's how the `Slugger` is implemented that would cause that error since each time it gets called the slug generated gets added to the Slug list to maintain uniqueness. 
Do we have a better way to handle this 🤔 ?

Reply:
The reason I bring up turning this into a remark plugin is that I believe this issue with the `Slugger` failing in dev mode is linked to the id processing being done in React, whereas in remark the whole page would get reprocessed with a new slugger instance if I understand correctly.

That said, that would be a pretty big refactor to fix a pretty niche edge case that would only ever show up in development, so no need to go for it right now.